File: Leetcode/41. First Missing Positive.java
import java.util.*;
class Solution {
    public int firstMissingPositive(int[] nums) {
        Set<Integer> set = new TreeSet<>();
        for(int i : nums) set.add(i);
        Integer new_nums[] = set.toArray(new Integer[0]);
        int num = 1;
        for(int i : new_nums){
            if(i <= 0) continue;
            if(i == num){
                num++;
            }
            else{
                return num;
            }
        };
        return num;
    }
}

File: Leetcode/1-999/41. First Missing Positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        nums = (i for i in nums if i>0)
        nums = list(set(nums))
        nums.sort()
        n = len(nums)
        i = 0
        if nums == []:
            return 1
        while (nums[i] == i+1):
            i += 1
            if i == n:
                return i+1
        else:
            return i+1
